api/magic_links/routes.py

```diff
@@ -89,7 +90,11 @@ def use(self, link_id: str):
             # else if no link exists (or it has been used)
             # but the user is already logged in
             # then redirect them to the global redirect url
-            frontend_account_url = f"{Config.MAGIC_LINK_REDIRECT_URL}?fund={fund_short_name}&round={round_short_name}"
+            frontend_account_url = (
+                f"{Config.MAGIC_LINK_REDIRECT_URL}"
+                f"?fund={urllib.parse.quote(fund_short_name)}"
+                f"&round={urllib.parse.quote(round_short_name)}"
+            )
```

Comment:
Could we use https://docs.python.org/3/library/urllib.parse.html#urllib.parse.urlencode to build the querystring instead?

Reply:
In this way we could strip the None values from the dict and pass it to the urlencode function.

Reply:
Ah! good point, happy to make this change https://github.com/communitiesuk/funding-service-design-authenticator/pull/210/files